plugins/webp-uploads/hooks.php

```diff
@@ -747,13 +747,112 @@ function webp_uploads_render_generator(): void {
+ *
+ * This function targets blocks like Cover and Group that may use background images,
+ * converting them to modern image formats when appropriate.
+ *
+ * @since n.e.x.t
+ *
+ * @phpstan-param array{
+ *                    blockName: string|null,
+ *                    attrs: array{
+ *                        id?: positive-int,
+ *                        url?: string,
+ *                        style?: array{
+ *                            background?: array{
+ *                                backgroundImage?: string
+ *                            }
+ *                        }
+ *                    }
+ *                } $block
+ *
+ * @param string|mixed         $block_content The block content.
+ * @param array<string, mixed> $block         The block.
+ * @return string The filtered block content.
+ */
+function webp_uploads_filter_block_background_images( $block_content, array $block ): string {
+	// Because plugins can do bad things.
+	if ( ! is_string( $block_content ) ) {
+		$block_content = '';
+	}
+
+	// Only run on frontend.
+	if ( ! webp_uploads_in_frontend_body() || '' === $block_content ) {
+		return $block_content;
+	}
+
+	$attachment_id = null;
+	$image_url     = '';
+
+	// Process Cover block.
+	if ( 'core/cover' === $block['blockName'] && isset( $block['attrs']['id'] ) ) {
+		$attachment_id = $block['attrs']['id'];
+		$image_url     = $block['attrs']['url'] ?? '';
+	}
+
+	// Process Group block with background image.
+	if ( 'core/group' === $block['blockName'] && isset( $block['attrs']['style']['background']['backgroundImage'] ) ) {
+		$bg_image = $block['attrs']['style']['background']['backgroundImage'];
+
+		if ( isset( $bg_image['id'] ) ) {
+			$attachment_id = $bg_image['id'];
+			$image_url     = $bg_image['url'] ?? '';
+		}
+	}
+
+	// Abort if there is no associated background image.
+	if ( is_null( $attachment_id ) || '' === $image_url || ! is_array( wp_get_attachment_metadata( $attachment_id ) ) ) {
+		return $block_content;
+	}
+
+	$original_mime = get_post_mime_type( $attachment_id );
+
+	if ( ! is_string( $original_mime ) ) {
+		return $block_content;
+	}
+
+	$target_mimes = webp_uploads_get_content_image_mimes( $attachment_id, 'background_image' );
+
+	foreach ( $target_mimes as $target_mime ) {
+		if ( $target_mime === $original_mime ) {
+			continue;
+		}
+
+		$new_url = webp_uploads_get_mime_type_image( $attachment_id, $image_url, $target_mime );
+
+		if ( is_string( $new_url ) ) {
+
+			$processor = new WP_HTML_Tag_Processor( $block_content );
+			while ( $processor->next_tag() ) {
+				$style = $processor->get_attribute( 'style' );
+
+				if ( is_string( $style ) && str_contains( $style, 'background-image:' ) && str_contains( $style, $image_url ) ) {
+					$updated_style = str_replace( $image_url, $new_url, $style );
+					$processor->set_attribute( 'style', $updated_style );
+					$block_content = $processor->get_updated_html();
+					break 2;
+				}
+			}
+		}
```

Comment:
For consistency with the use of `continue` above.

```suggestion

		if ( ! is_string( $new_url ) ) {
			continue;
		}

		$processor = new WP_HTML_Tag_Processor( $block_content );
		while ( $processor->next_tag() ) {
			$style = $processor->get_attribute( 'style' );

			if ( is_string( $style ) && str_contains( $style, 'background-image:' ) && str_contains( $style, $image_url ) ) {
				$updated_style = str_replace( $image_url, $new_url, $style );
				$processor->set_attribute( 'style', $updated_style );
				$block_content = $processor->get_updated_html();
				break 2;
			}
		}
```